src/lib.rs

```diff
@@ -891,12 +892,22 @@ impl<'a> SeedableRng<&'a [usize]> for StdRng {
 /// seeded `Rng` for consistency over time you should pick one algorithm and
 /// create the `Rng` yourself.
 ///
-/// This will read randomness from the operating system to seed the
-/// generator.
+/// This will seed the generator with randomness from the operating system or,
+/// if the operating system returns an error, the current system time.
 pub fn weak_rng() -> XorShiftRng {
     match OsRng::new() {
         Ok(mut r) => r.gen(),
-        Err(e) => panic!("weak_rng: failed to create seeded RNG: {:?}", e)
+        Err(_) => {
+            // Set a bit because XorShiftRng will panic if the seed is
+            // entirely zero.
```

Comment:
Are you sure you don't *want* this to panic if `precise_time_ns` is so broken it returns zero? You've got zero entropy in this case... for *some* uses that doesn't matter but not necessarily all.